src/event_handlers.py

```diff
@@ -0,0 +1,140 @@
+# pylint: disable=too-few-public-methods
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Union
+
+from celery import Task
+from loguru import logger
+
+from src.constants import EventEnum, EventType, LabelName
+from src.helpers import get_exception_class
+from src.instrumentation import EventCounter, EventGauge
+
+
+class InvalidTaskEventLabelName(Exception):
+    pass
+
+
+class IEventHandler(ABC):
+    def __init__(self, state):
+        self._state = state
+
+    def __call__(self, event: Dict[str, Any]):
```

Comment:
Convenience method so that once we define an instance of the handler we can just pass it as an event handler to the receiver.